### PR TITLE
Support Travis CI testing

### PR DESCRIPTION
**Pull request by jslee02**
This pull request enables to perform build tests for FCL on [Travis CI](https://travis-ci.org). To make this work, GitHub Webhook should be activated by one of the maintainers of this repository. Here is a [short instruction](http://docs.travis-ci.com/user/getting-started/#Step-two%3A-Activate-GitHub-Webhook).

To test on Mac, [we need to send an request email to Travis-CI](http://blog.travis-ci.com/2014-05-13-multi-os-feature-available/).

Some tests take too much time on debug mode for Travis-CI. Travis-CI regards a test is failed if there is no response for more than 10 minutes. I reduced the size of the tests not to take more than 10 minutes.

There were some broken tests (see #25), and they are disabled for now. These broken tests will be tracked in #25.

### CODE REVIEW

**Comment by scpeters** [comment]
I just enabled travis for this project. @jslee02 if you close and re-open this, then it should trigger a rebuild.

**Comment by jslee02** [comment]
Great, the first build test has been passed!

To enable testing on osx, someone need to send an request email to Travis-CI team. I'm not sure if it's acceptable to send the request email by who has no write access.

**Comment by scpeters** [comment]
From that [page linked above](http://blog.travis-ci.com/2014-05-13-multi-os-feature-available/):

"Update: We are not currently accepting multi-OS requests

We are at capacity with Mac workers in both .org and .com environments. Consequently, we are not accepting requests for the multi-OS feature at this time. As more capacity is added, we will resume accepting requests."

So I think we'll have to wait on adding travis OS X support.

**Comment by scpeters** [comment]
:+1:

File: include/fcl/config.h.in [hunk 1]
```diff
@@ -45,4 +45,7 @@
 #cmakedefine01 FCL_HAVE_FLANN
 #cmakedefine01 FCL_HAVE_TINYXML
 
+#cmakedefine01 BUILD_TYPE_DEBUG
+#cmakedefine01 BUILD_TYPE_RELEASE
```
Comment (scpeters) [inline include/fcl/config.h.in]: Hm...maybe we should prefix with `FCL_`? ie. `FCL_BUILD_TYPE_DEBUG`, `FCL_BUILD_TYPE_RELEASE`?

Comment (jslee02) [inline include/fcl/config.h.in]: It looks consistent naming. Changed in 26dfacf.

**Comment by scpeters** [comment]
:+1:

**Comment by scpeters** [comment]
I'll give it a day for further comments, then merge.

**Pull request merged**